### PR TITLE
ci: go back to stricter checks

### LINKED ISSUE #1006: Warning from R CMD check
```r
W  checking whether package ‘igraph’ can be installed (2m 7.3s)
   Found the following significant warnings:
     rinterface_extra.c:3149:3: warning: ‘igraph_incidence’ is deprecated [-Wdeprecated-declarations]
     rinterface_extra.c:3261:34: warning: passing argument 5 of ‘igraph_sparsemat_view’ from incompatible pointer type [-Wincompatible-pointer-types]
     rinterface_extra.c:3261:53: warning: passing argument 6 of ‘igraph_sparsemat_view’ from incompatible pointer type [-Wincompatible-pointer-types]
     rinterface_extra.c:4416:3: warning: ‘igraph_lattice’ is deprecated [-Wdeprecated-declarations]
     rinterface_extra.c:5004:3: warning: ‘igraph_erdos_renyi_game’ is deprecated [-Wdeprecated-declarations]
     rinterface_extra.c:6479:3: warning: ‘igraph_read_graph_dimacs’ is deprecated [-Wdeprecated-declarations]
     rinterface_extra.c:6537:3: warning: ‘igraph_write_graph_dimacs’ is deprecated [-Wdeprecated-declarations]
     rinterface_extra.c:8327:3: warning: ‘igraph_laplacian’ is deprecated [-Wdeprecated-declarations]
     rinterface.c:3853:3: warning: ‘igraph_hub_score’ is deprecated [-Wdeprecated-declarations]
     rinterface.c:3902:3: warning: ‘igraph_authority_score’ is deprecated [-Wdeprecated-declarations]
     rinterface.c:5300:3: warning: ‘igraph_random_edge_walk’ is deprecated [-Wdeprecated-declarations]
     rinterface.c:5927:3: warning: ‘igraph_bipartite_game’ is deprecated [-Wdeprecated-declarations]
     rinterface.c:9970:3: warning: ‘igraph_subisomorphic_function_vf2’ is deprecated [-Wdeprecated-declarations]
     rinterface.c:11829:3: warning: ‘igraph_get_stochastic_sparsemat’ is deprecated [-Wdeprecated-declarations]
     rinterface.c:11853:3: warning: ‘igraph_hrg_dendrogram’ is deprecated [-Wdeprecated-declarations]
     vendor/cigraph/src/core/matrix.c:234:13: warning: ‘igraph_vector_e_tol’ is deprecated [-Wdeprecated-declarations]
   See ‘/home/maelle/Documents/cynkra/igraph.Rcheck/00install.out’ for details.
```

@krlmlr is this expected?

### PR DESCRIPTION
**Pull request by maelle**
Fix #1006

Now I understand why everything was green on CI!

### CODE REVIEW

**Comment by maelle** [comment]
I think it'd make sense to merge then rebase all PRs that work towards getting the build green again?

**Review by krlmlr — APPROVED** [review]
Thanks! Should be good now.

**Comment by krlmlr** [comment]
@Antonov548: We now have compilation warnings on a few platforms only, e.g., on macOS:

```
  rinterface.c:6349:84: warning: variable 'c_outfile' is uninitialized when used here [-Wuninitialized]
  rinterface.c:8381:58: warning: variable 'c_instream' is uninitialized when used here [-Wuninitialized]
  rinterface.c:8441:59: warning: variable 'c_outstream' is uninitialized when used here [-Wuninitialized]
  rinterface.c:10886:33: warning: variable 'c_a' is uninitialized when used here [-Wuninitialized]
  rinterface.c:10886:38: warning: variable 'c_b' is uninitialized when used here [-Wuninitialized]
  rinterface.c:10886:43: warning: variable 'c_eps' is uninitialized when used here [-Wuninitialized]
  rinterface.c:10910:31: warning: variable 'c_a' is uninitialized when used here [-Wuninitialized]
  rinterface.c:10910:36: warning: variable 'c_b' is uninitialized when used here [-Wuninitialized]
  rinterface.c:10910:41: warning: variable 'c_eps' is uninitialized when used here [-Wuninitialized]
  rinterface.c:10917:10: warning: variable 'r_result' is uninitialized when used here [-Wuninitialized]
  rinterface.c:11610:10: warning: variable 'r_result' is uninitialized when used here [-Wuninitialized]
  rinterface.c:11632:10: warning: variable 'r_result' is uninitialized when used here [-Wuninitialized]
  rinterface.c:11646:28: warning: variable 'c_igraph_errno' is uninitialized when used here [-Wuninitialized]
  rinterface.c:11653:10: warning: variable 'r_result' is uninitialized when used here [-Wuninitialized]
  rinterface.c:11761:31: warning: variable 'version_string' is uninitialized when used here [-Wuninitialized]
  rinterface.c:11762:31: warning: variable 'major' is uninitialized when used here [-Wuninitialized]
  rinterface.c:11763:31: warning: variable 'minor' is uninitialized when used here [-Wuninitialized]
  rinterface.c:11764:31: warning: variable 'subminor' is uninitialized when used here [-Wuninitialized]
```

Is this dead code? We can remove in `rinterface.c` and reconcile later, we have drift anyway.

There are more failures on Windows.

Can you please push to this branch?

**Comment by Antonov548** [comment]
> @Antonov548: We now have compilation warnings on a few platforms only, e.g., on macOS:
> 
> ```
>   rinterface.c:6349:84: warning: variable 'c_outfile' is uninitialized when used here [-Wuninitialized]
>   rinterface.c:8381:58: warning: variable 'c_instream' is uninitialized when used here [-Wuninitialized]
>   rinterface.c:8441:59: warning: variable 'c_outstream' is uninitialized when used here [-Wuninitialized]
>   rinterface.c:10886:33: warning: variable 'c_a' is uninitialized when used here [-Wuninitialized]
>   rinterface.c:10886:38: warning: variable 'c_b' is uninitialized when used here [-Wuninitialized]
>   rinterface.c:10886:43: warning: variable 'c_eps' is uninitialized when used here [-Wuninitialized]
>   rinterface.c:10910:31: warning: variable 'c_a' is uninitialized when used here [-Wuninitialized]
>   rinterface.c:10910:36: warning: variable 'c_b' is uninitialized when used here [-Wuninitialized]
>   rinterface.c:10910:41: warning: variable 'c_eps' is uninitialized when used here [-Wuninitialized]
>   rinterface.c:10917:10: warning: variable 'r_result' is uninitialized when used here [-Wuninitialized]
>   rinterface.c:11610:10: warning: variable 'r_result' is uninitialized when used here [-Wuninitialized]
>   rinterface.c:11632:10: warning: variable 'r_result' is uninitialized when used here [-Wuninitialized]
>   rinterface.c:11646:28: warning: variable 'c_igraph_errno' is uninitialized when used here [-Wuninitialized]
>   rinterface.c:11653:10: warning: variable 'r_result' is uninitialized when used here [-Wuninitialized]
>   rinterface.c:11761:31: warning: variable 'version_string' is uninitialized when used here [-Wuninitialized]
>   rinterface.c:11762:31: warning: variable 'major' is uninitialized when used here [-Wuninitialized]
>   rinterface.c:11763:31: warning: variable 'minor' is uninitialized when used here [-Wuninitialized]
>   rinterface.c:11764:31: warning: variable 'subminor' is uninitialized when used here [-Wuninitialized]
> ```
> 
> Is this dead code? We can remove in `rinterface.c` and reconcile later, we have drift anyway.
> 
> There are more failures on Windows.
> 
> Can you please push to this branch?

Would we like to keep synchronized generated code with `functions.yml`?

**Comment by krlmlr** [comment]
We reconcile at some point, yes, but for now, drift is fine. I want to move
to autogenerating everything, any Stimulus changes might have to be
reverted anyway.

We can edit interface.c and aaa-auto.R directly until after the release.

Michael Antonov ***@***.***> schrieb am Fr. 12. Jan. 2024 um
16:06:

**Comment by szhorvat** [comment]
Beware that most of these "uninitialized when used here" warning indicate actual bugs! If the affected function is used, that's a problem we need to fix now. Hopefully none of them are used?

**Comment by krlmlr** [comment]
Now there's only Windows and macOS failing, for good reason I think.

**Comment by Antonov548** [comment]
> Now there's only Windows and macOS failing, for good reason, I think.

Yes, checking

File: src/rinterface_extra.c [hunk 1]
```diff
@@ -73,8 +73,8 @@ enum igraph_versions {
 void R_check_int_scalar(SEXP value)
 {
   if (Rf_xlength(value) != 1) {
-    igraph_errorf("Expecting a scalar integer but received a vector of length %zu.",
-                    __FILE__, __LINE__, IGRAPH_EINVAL, (size_t) Rf_xlength(value));
+    igraph_errorf("Expecting a scalar integer but received a vector of length %d.",
+                    __FILE__, __LINE__, IGRAPH_EINVAL, Rf_xlength(value));
```
Comment (szhorvat) [inline src/rinterface_extra.c:77]: Why did you make this change? `%d` is only for `int`, and the return type of `Rf_xlength()` is system-dependent (may not be compatible with `int`!), but _always fits in `size_t`_. I chose `%zu` and a cast to `size_t` in purpose.

Comment (Antonov548) [reply src/rinterface_extra.c:77]: `zu` is not compatible with some versions of compiler. Maybe good solution will be to use `lu`?

Comment (szhorvat) [reply src/rinterface_extra.c]: What specific compiler is it not compatible with?

Comment (szhorvat) [reply src/rinterface_extra.c]: `zu` is standard in both C99 and C++11. Could it be that the proper standard is not set? I'd be very surprised to see any still-relevant compiler not support it.

Comment (szhorvat) [reply src/rinterface_extra.c]: Works with Clang 3.4 and GCC 4.8. igraph itself wouldn't support anything earlier.

Comment (Antonov548) [reply src/rinterface_extra.c]: > According to tests wit Godbolt, works with Clang 3.4 and GCC 4.8. igraph itself wouldn't support anything earlier.

Here: https://github.com/igraph/rigraph/actions/runs/7512815244/job/20453943297#step:9:84
Maybe for windows there is really wrong flags. I will check.

Comment (szhorvat) [reply src/rinterface_extra.c]: Can you check what was the precise compilation command? Was there an `-std` flag?

**Comment by maelle** [comment]
@Antonov548 thanks! What are the windows and sanitizer failures?

File: src/rinterface_extra.c (hunk 1 above)
Comment (szhorvat) [reply src/rinterface_extra.c]: I see you are casting to `long int`, but `%lu` is actually `unsigned long`.

But please do check what the compilation command is. `%zu` has been part of the standard for a very long time. There is no way that the compiler does not support it. More likely, the wrong standard was chosen.

Can you try separately with the same GCC version? Does it accept `%zu`?

Comment (Antonov548) [reply src/rinterface_extra.c]: > I see you are casting to `long int`, but `%lu` is actually `unsigned long`.
> 
> But please do check what the compilation command is. `%zu` has been part of the standard for a very long time. There is no way that the compiler does not support it. More likely, the wrong standard was chosen.
> 
> Can you try separately with the same GCC version? Does it accept `%zu`?

Oh, yes. Regarding `int` it's typo.

I checked the compile flags. And also added `-std=c99`, but I didn't help. Still see the warning:
```gcc  -I"C:/PROGRA~1/R/R-43~1.1/include" -DNDEBUG -I"C:/rtools43/x86_64-w64-mingw32.static.posix/include/libxml2" -I"C:/rtools43/x86_64-w64-mingw32.static.posix/include" -DLIBXML_STATIC -DUSING_R -I. -Ivendor -Ivendor/cigraph/src -Ivendor/cigraph/include -Ivendor/cigraph/vendor -DNDEBUG -DNTIMER -DNPRINT -DINTERNAL_ARPACK -DIGRAPH_THREAD_LOCAL= -DPRPACK_IGRAPH_SUPPORT -D_GNU_SOURCE=1 -std=c99  -I"C:/rtools43/x86_64-w64-mingw32.static.posix/include"     -O2 -Wall  -mfpmath=sse -msse2 -mstackrealign  -c rinterface_extra.c -o rinterface_extra.o```

using C compiler: 'gcc.exe (GCC) 12.3.0'.

Comment (Antonov548) [reply src/rinterface_extra.c]: @szhorvat https://stackoverflow.com/questions/68900199/how-to-get-mingw-gcc-to-recognize-the-zu-format-specifier-for-size-t

Hm, looks like there still no agreement for `zu`.

**Comment by Antonov548** [comment]
> @Antonov548 thanks! What are the windows and sanitizer failures?

On windows there is warning in `vendor` folder, precise in core igraph. Not sure why the are not ignored. I though that vendor folder is ignored.

**Comment by maelle** [comment]
True, `^vendor$` is listed in `.Rbuildignore` so they should be ignored :thinking:

File: src/rinterface_extra.c (hunk 1 above)
Comment (szhorvat) [reply src/rinterface_extra.c]: What that SO answer is saying is that it does actually work but MinGW's GCC _incorrectly_ warns about it. That is very disappointing.

You should be aware that on Windows the `long` type is 32-bit long. Using `long` here is actually wrong. Can't we just use `-Wno-format` specifically on Windows?

Comment (szhorvat) [reply src/rinterface_extra.c]: @Antonov548 Can you try to use `PRIuPTR` instead? Here is an example:

https://godbolt.org/z/eMjM3nz9b

**Comment by szhorvat** [comment]
@Antonov548 As a general rule, never use `long` in igraph code. `long` has an unpredictable width, and _will_ cause trouble on Windows (where it is 32-bit).

File: src/rinterface_extra.c (hunk 1 above)
Comment (Antonov548) [reply src/rinterface_extra.c]: Seems `uintptr_t` works well. Let's see in ci. 
I guess it's better to leave all warning enabled.

Comment (szhorvat) [reply src/rinterface_extra.c]: Yes, it's better to leave warnings enabled on all platforms.

Someone just told me about the `__USE_MINGW_ANSI_STDIO` macro. Defining this to either `1` or `0` may help (unclear).

But if `PRIuPTR` works, let's just stick to that, as it is standard.

**Pull request merged**